addons/voice/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/protocol/AudioConfig.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.voice.googletts.internal.protocol;
+
+/**
+ * The configuration of the synthesized audio.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+public class AudioConfig {
+
+    /**
+     * Required. The format of the requested audio byte stream.
+     */
+    private AudioEncoding audioEncoding;
+
+    /**
+     * Optional speaking pitch, in the range [-20.0, 20.0]. 20 means increase 20 semitones from the original pitch. -20
+     * means decrease 20 semitones from the original pitch.
+     */
+    private Double pitch;
+
+    /**
+     * The synthesis sample rate (in hertz) for this audio. Optional. If this is different from the voice's natural
+     * sample rate, then the synthesizer will honor this request by converting to the desired sample rate (which might
+     * result in worse audio quality), unless the specified sample rate is not supported for the encoding chosen, in
+     * which case it will fail the request and return google.rpc.Code.INVALID_ARGUMENT.
+     */
+    private Long sampleRateHertz;
+
+    /**
+     * Optional speaking rate/speed, in the range [0.25, 4.0]. 1.0 is the normal native speed supported by the specific
+     * voice. 2.0 is twice as fast, and 0.5 is half as fast. If unset(0.0), defaults to the native 1.0 speed. Any other
+     * values < 0.25 or > 4.0 will return an error.
+     */
+    private Double speakingRate;
+
+    /**
+     * Optional volume gain (in dB) of the normal native volume supported by the specific voice, in the range [-96.0,
+     * 16.0]. If unset, or set to a value of 0.0 (dB), will play at normal native signal amplitude. A value of -6.0 (dB)
+     * will play at approximately half the amplitude of the normal native signal amplitude. A value of +6.0 (dB) will
+     * play at approximately twice the amplitude of the normal native signal amplitude. Strongly recommend not to exceed
+     * +10 (dB) as there's usually no effective increase in loudness for any value greater than that.
+     */
+    private Double volumeGainDb;
+
+    public AudioConfig() {
```

Comment:
Do we need these empty constructors in the POJO classes?

Reply:
AFAIK they are used by Gson when deserializing objects from JSON.

Reply:
AFAIK in newer versions of Gson, you do not even have to declare a default constructor anymore.

But as long as the Gson user guide states that we should define a default no-args constructor for any class to work with Gson properly we should follow their recommendations.